src/appendix-code-index.md

```diff
@@ -7,6 +7,10 @@ compiler.
 Item            |  Kind    | Short description           | Chapter            | Declaration
 ----------------|----------|-----------------------------|--------------------|-------------------
 `CodeMap` | struct | The CodeMap maps the AST nodes to their source code | [The parser](the-parser.html) | [src/libsyntax/codemap.rs](https://github.com/rust-lang/rust/blob/master/src/libsyntax/codemap.rs)
+`CompileState` | struct | State that is passed to a callback at each compiler pass | [The Rustc Driver](rustc-driver.html) | [src/librustc_driver/driver.rs](https://github.com/rust-lang/rust/blob/master/src/librustc_driver/driver.rs)
+`ast::Crate` | struct | Syntax-level representation of a parsed crate | | [src/librustc/hir/mod.rs](https://github.com/rust-lang/rust/blob/master/src/libsyntax/ast.rs)
+`hir::Crate` | struct | Top-level data structure representing the crate being compiled | | [src/librustc/hir/mod.rs](https://github.com/rust-lang/rust/blob/master/src/librustc/hir/mod.rs)
```

Comment:
"Top-level" isn't very helpful here - the HIR is basically a compiler-internal version of the AST, i.e., tools and other users should not use it (but can use the AST). It does not match the source text as closely as the AST, being more designed for compiler use, but still fits the 'textbook' definition of an AST.

Reply:
Would it make sense to say "more compiler friendly" form of the AST? I believe the HIR is a desugared, more useful form of AST where we've broken things out into their categories (items, trait impls, etc).